src/main/java/com/commercetools/sync/commons/utils/CompletableFutureUtils.java

```diff
@@ -0,0 +1,21 @@
+package com.commercetools.sync.commons.utils;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class CompletableFutureUtils {
+
+    private static final CompletableFuture<Optional<?>> EMPTY = CompletableFuture.completedFuture(Optional.empty());
+
+    /**
+     * Returns a {@link CompletableFuture} of an empty {@link Optional}. There is no guarantee
+     * that it is a singleton.
```

Comment:
> here is no guarantee that it is a singleton.

what does this mean?

Reply:
From the first glance it looks like a singleton. But actually it is not. 
So I just stole a comment from the [Optional#empty()](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#empty--) JavaDoc

Reply:
don't think we have to mention it here, but even if we decided to make such a comment - pls, format them as in that example, because it is less confusing.